src/main/java/org/jabref/logic/bibtex/DuplicateCheck.java

```diff
@@ -98,7 +102,16 @@ public static boolean isDuplicate(BibEntry one, BibEntry two, BibDatabaseMode bi
         return req[0] >= DuplicateCheck.duplicateThreshold;
     }
 
-    private static boolean hasSameIdentifier(BibEntry one, BibEntry two) {
+    private static boolean haveDifferentEditions(BibEntry one, BibEntry two) {
+        if (one.getField(FieldName.EDITION).isPresent() && two.getField(FieldName.EDITION).isPresent()) {
+            if (!one.getField(FieldName.EDITION).get().equals(two.getField(FieldName.EDITION).get())) {
```

Comment:
You can direclty compare Optionals with equals, but be aware that two empty optionals also equal.
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#equals-java.lang.Object-